### PR TITLE
Fix wB97X-D3 and add KMLYP

### LINKED ISSUE #2206: Incorrect functional parameters for wB97X-D3
Hello,

I'm a developer at Schrodinger working on our QM engine, Jaguar. Recently I implemented wB97X-D3 and was using PSI4 to check my results. Interestingly, I found discrepancies in our SCF energies (difference of 0.003 Ha on a methane molecule), but the dispersion corrections matched to 1e-6. 

After some investigation, I believe PSI4's wB97X-D3 functional is using the wrong parameters for the functional (the dispersion correction seems fine). 

If you look in the output file, you can see 

>  ==> DFT Potential <==
>
>  => Composite Functional: WB97X-D3ZERO <= 
>
>    wB97X-d3zero Hyb-GGA Exchange-Correlation Functional
>
>    J.-D. Chai and M. Head-Gordon, J. Chem. Phys. 128, 084106 (2008)
>
> ....
>   => Exchange-Correlation Functionals <=
>
>    1.0000   wB97X range-separated functional
>
>   => Exact (HF) Exchange <=
>
>    0.8423            HF,LR [omega = 0.3000]
>    0.1577               HF 

The HF exchange and omega printouts indicate that you are using wB97X parameters in the functional. But note that wB97X and wB97X-D3 differ in their functional parameters (for instance in wB97X-D3, omega=0.25. Other parameters differ as well. See Chai et al, dx.doi.org/10.1021/ct300715s, Table 1 for the full list)

I did some testing by substituting wB97X parameters into Jaguar's wB97X-D3 call. Indeed, Jaguar then returns matching SCF energies with PSI4's wB97X-D3. I was comparing to PSI4 version: Psi4 1.4a2.dev723, Git: Rev {master} fb499f4

If you guys have fixed this in a newer version of PSI4, then great! If not, perhaps the parameters could be updated to match the paper? Of course, if I'm mistaken please let me know as well.

Cheers,
Alan Chien

### PR DESCRIPTION
**Pull request by PeterKraus**
## Description
Fixed wB97X-D3 to use the correct libxc functional, added KMLYP, and added/modified tests for both. Fixes #2206.

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Review by hokru — APPROVED** [review]
Thanks for the quick fix Peter!

File: tests/pytests/test_dft_benchmarks.py [hunk 1]
```diff
@@ -154,11 +154,12 @@ def name_dft_test(val):
     pytest.param(        'wB97M-V', 0.4544676075, '6-31G'),  # Q-Chem
     pytest.param(          'wB97X', 0.4564711283, '6-31G'),  # Q-Chem
     pytest.param(        'wB97X-D', 0.4575912358, '6-31G', marks=pytest.mark.quick),  # Q-Chem
-    pytest.param(       'wB97X-D3', 0.4570744381, '6-31G', marks=[using("dftd3"), pytest.mark.xfail]),  # Q-Chem  # needs tweaks in LibXC
+    pytest.param(       'wB97X-D3', 0.4570744381, '6-31G', marks=[using("dftd3")]),   # Q-Chem  # needs tweaks in LibXC
```
Comment (JonathonMisiewicz) [inline tests/pytests/test_dft_benchmarks.py:157]: Is "needs tweaks in LibXC" still correct?

Comment (PeterKraus) [reply tests/pytests/test_dft_benchmarks.py]: Nope!

**Review by JonathonMisiewicz — APPROVED** [review]

**Review by jturney — APPROVED** [review]
That was quick!

**Review by loriab — APPROVED** [review]
Ack! Functional definition and dispersion definition were so cleanly separated in my mind :-(

Thanks for the detection and the fix! In case someone else had the same question, I'm not finding a D3(BJ) disp set defined for wB97X.

**Comment by PeterKraus** [comment]
Oh, wB97X-D3(BJ) exists, but it's based on the wB97X-V with -D3(BJ) dispersion.

**Pull request merged**